lib/puppet/application/lookup.rb

```diff
@@ -331,6 +330,18 @@ def generate_scope
     end
 
     node = Puppet::Node.indirection.find(node) unless node.is_a?(Puppet::Node) # to allow unit tests to pass a node instance
+
+    if options[:fact_file]
+      original_facts = node.facts.values
+      if options[:fact_file].match(/.*\.json/)
+        given_facts = JSON.parse(File.read(options[:fact_file]))
+      else
+        given_facts = YAML.load(File.read(options[:fact_file]))
+      end
```

Comment:
Both of the above paths an fail on faulty data, missing file etc. When that happens does those errors surface and give intelligent feedback to the user, or should the errors be rescued and reported?